app/controllers/admins_controller.rb

```diff
@@ -102,19 +107,32 @@ def uploadRoster
                 # Add this user to the course
                 # Look for this user
                 email = newCUD[:email]
+                first_name = newCUD[:first_name]
+                last_name = newCUD[:last_name]
+                school = newCUD[:school]
+                major = newCUD[:major]
+                year = newCUD[:year]
                 
                 if ((user = User.where(email: email).first).nil?)
                   # Create a new user
-                  user = User.roster_create(email, newCUD[:first_name],
-                                            newCUD[:last_name])
+                  user = User.roster_create(email, first_name, last_name, school,
+                                        major, year)
                   if (user.nil?)
                     throw NewUserCreationException
                   end
+                else
+                  # Override current user
+                  user.first_name = first_name
+                  user.last_name = last_name
+                  user.school = school
+                  user.major = major
+                  user.year = year
+                  user.save
                 end
                 
                 # Make sure this user doesn't have a cud in the course
                 if (@course.course_user_data.where(user: user).first)
-                  throw NewUserExistInCourseException
+                  throw GreenCUDExistInCourseException
```

Comment:
And here.  (As a side note, Ruby has a raise and a throw? WTF? What hasn't this language absorbed from every other language ever?)